### ISSUE TITLE
pm2 is broken because childprocess.unref() fails

### ISSUE THREAD

**Issue by khrj** [denoland/deno#16784]
![image](https://user-images.githubusercontent.com/44947946/203769851-7f71c3c6-d8a7-46c2-bb5d-79c109e31bfa.png)

When running with:
- [`__proto__` enabled](https://gist.github.com/khrj/e7a33c70fabb1c0430745842e45950ba) (applied on top of [the changes in #16775](https://github.com/denoland/deno/compare/main..f8d5177#diff-00a8b60a7dc00e07ca154218d7c1f478aedfa44a0b6055e558c1463ca1515dbb))
- PR for pm2 to not use `__proto__`: https://github.com/Unitech/pm2/pull/5489

`pm2` seems to fail because `child.unref()` failed internally

**Comment by khrj** [comment]
Note that this is only reproduced when running pm2 for the first time after startup (when it shows `[PM2] Spawning PM2 daemon with pm2_home`). After that, you need to `deno run --unstable -A npm:pm2 kill` to reproduce it again

**Comment by khrj** [comment]
Possibly related missing implementation:

```js
import { createRequire } from "https://deno.land/std@0.165.0/node/module.ts"
const require = createRequire(import.meta.url)

const child_process = require("child_process")
const fs = require("fs")

const pipe = fs.openSync("/tmp/issue.log", 'a')

const child = child_process.spawn("echo", {
	stdio      : ['ipc', pipe, pipe]
})
```
![image](https://user-images.githubusercontent.com/44947946/203779137-2d457644-d21a-4416-8ef6-850c458d40e8.png)

**Comment by bartlomieju** [comment]
Unrefing child process is definitely implemented (eg. Vite uses it), but the problem seems to be usage of `ipc` as stdio.

**Comment by littledivy** [comment]
child_process IPC is now implemented. See https://github.com/denoland/deno/pull/21490

**Comment by bartlomieju** [comment]
This issue should be resolved as of Deno v1.39.1.
